GitUI/ScriptsEngine/ScriptOptionsParser.cs

```diff
@@ -106,6 +110,27 @@ public static (string? arguments, bool abort) Parse(string? arguments, IGitUICom
             List<IGitRef> currentBranches = new();
             List<IGitRef> currentTags = new();
 
+            // Specific handling of "UserInput" because the value entered should not replace all the variables
+            Match match;
+            while ((match = userInputRegex.Match(arguments)).Success)
+            {
+                Group defaultValueMatch = match.Groups["defaultValue"];
+                (string? arguments, bool abort) defaultValue = defaultValueMatch == null ? (string.Empty, false) : Parse(defaultValueMatch.Value ?? string.Empty, uiCommands, owner, scriptHostControl);
+                if (defaultValue.abort)
+                {
+                    return (arguments: null, abort: true);
+                }
+
+                string label = match.Groups["label"].Value;
+
+                using (IUserInputPrompt prompt = uiCommands.GetService<ISimplePromptCreator>().Create(label, defaultValue.arguments))
+                {
+                    prompt.ShowDialog(owner);
+                    arguments = ReplaceOption($"UserInput:{label}", arguments, prompt.UserInput);
+                    arguments = ReplaceOption(match.Value.Substring(1, match.Value.Length - 2), arguments, prompt.UserInput);
```

Comment:
❔ Should we support quoting by double braces, too? Or is it handled by `ReplaceOption` already?

Reply:
It is handled inside `ReplaceOption()`.